src/components/app/ProfileSharing.js

```diff
@@ -224,6 +241,14 @@ class ProfileSharingCompositeButton extends PureComponent {
         if (this._uploadErrorButton) {
           this._uploadErrorButton.openPanel();
         }
+        const { ga } = window;
+        if (ga) {
+          ga('send', {
+            hitType: 'event',
+            eventCategory: 'profile upload',
+            eventAction: 'failed',
+          });
+        }
```

Comment:
I'd create an action (I mean, an action that wouldn't dispatch a redux action) to handle ga calls.

or at least a local function.

Reply:
I'm curious if the reason you want me to create local functions is the if check. Would you be ok with me stubbing out the analytics function for everywhere? That way we can fire it regardless. The interface itself is just a function call with some parameters, so I don't feel the need to abstract it away further.